examples/tiny.c

```diff
@@ -62,87 +73,90 @@ usage( const char *program ) {
            program, program );
 }
 
-coap_context_t *
-get_context(const char *node, const char *port) {
-  coap_context_t *ctx = NULL;
+static coap_session_t *
+get_session(const char *group) {
+  coap_context_t *ctx = coap_new_context(NULL);
   int s;
   struct addrinfo hints;
   struct addrinfo *result, *rp;
+  coap_session_t *session;
+  int hops = 16;
+
+  if (!ctx)
+    return NULL;
 
   memset(&hints, 0, sizeof(struct addrinfo));
   hints.ai_family = AF_UNSPEC;    /* Allow IPv4 or IPv6 */
   hints.ai_socktype = SOCK_DGRAM; /* Coap uses UDP */
   hints.ai_flags = AI_PASSIVE | AI_NUMERICHOST | AI_NUMERICSERV | AI_ALL;
 
-  s = getaddrinfo(node, port, &hints, &result);
+  s = getaddrinfo(group, NULL, &hints, &result);
   if ( s != 0 ) {
     fprintf(stderr, "getaddrinfo: %s\n", gai_strerror(s));
     return NULL;
   }
 
   /* iterate through results until success */
   for (rp = result; rp != NULL; rp = rp->ai_next) {
-    ctx = coap_new_context(rp->ai_addr, rp->ai_addrlen);
-    if (ctx) {
-      /* TODO: output address:port for successful binding */
-      goto finish;
+    coap_address_t addr;
+    coap_address_init(&addr);
+    addr.size = rp->ai_addrlen;
+    memcpy(&addr.addr, rp->ai_addr, rp->ai_addrlen);
+
+    session = coap_new_client_session(ctx, NULL, &addr, COAP_PROTO_UDP);
+    if (!session)
+      continue;
+
+    if (IN6_IS_ADDR_MULTICAST(&addr.addr.sin6.sin6_addr) ) {
+      /* set socket options for multicast */
+      if ( setsockopt(session->sock.fd, IPPROTO_IPV6, IPV6_MULTICAST_HOPS,
+                       (char *)&hops, sizeof(hops) ) < 0 )
+        perror("setsockopt: IPV6_MULTICAST_HOPS");
+
     }
+    freeaddrinfo(result);
+    return session;
   }
 
-  fprintf(stderr, "no context available for interface '%s'\n", node);
-
- finish:
+  fprintf(stderr, "no session available for group '%s'\n", group);
   freeaddrinfo(result);
-  return ctx;
+  return NULL;
 }
 
 int
 main(int argc, char **argv) {
-  coap_context_t  *ctx;
   struct timeval tv;
   coap_pdu_t  *pdu;
-  struct sockaddr_in6 dst;
-  int hops = 16;
+  coap_session_t *session;
 
   if ( argc > 1 && strncmp(argv[1], "-h", 2) == 0 ) {
     usage( argv[0] );
     exit( 1 );
   }
 
-  ctx = get_context("::", NULL);
-  if ( !ctx )
+  session = get_session(argc > 1 ? argv[1] : "::1");
+
+  if ( !session )
     return -1;
 
   id = rand() & INT_MAX;
 
-  memset(&dst, 0, sizeof(struct sockaddr_in6 ));
-  dst.sin6_family = AF_INET6;
-  inet_pton( AF_INET6, argc > 1 ? argv[1] : "::1", &dst.sin6_addr );
-  dst.sin6_port = htons( COAP_DEFAULT_PORT );
-
-  if ( IN6_IS_ADDR_MULTICAST(&dst.sin6_addr) ) {
-    /* set socket options for multicast */
-
-    if ( setsockopt( ctx->sockfd, IPPROTO_IPV6, IPV6_MULTICAST_HOPS,
-                     (char *)&hops, sizeof(hops) ) < 0 )
-      perror("setsockopt: IPV6_MULTICAST_HOPS");
-
-  }
+  signal(SIGINT, handle_sigint);
```

Comment:
For consistency (and portability, although this is not that important for this example), I suggest to use `sigaction()` here as well.

Reply:
Fixed